src/as-pool.c

```diff
@@ -2213,6 +2213,33 @@ as_pool_get_components_by_extends (AsPool *pool, const gchar *extended_id)
 	return result;
 }
 
+/**
+ * as_pool_get_components_by_bundle_id: (skip)
+ * @pool: An instance of #AsPool.
+ * @kind: The kind of the bundle we are looking for
+ * @bundle_id: The ID, as specified in AsBundle
+ *
+ * Find components that provide a bundle with a specific id.
+ *
+ * Since: 0.16.0
+ */
+GPtrArray*
+as_pool_get_components_by_bundle_id (AsPool *pool, AsBundleKind kind, const gchar *bundle_id)
+{
+	AsPoolPrivate *priv = GET_PRIVATE (pool);
+	g_autoptr(GError) tmp_error = NULL;
+	GPtrArray *result = NULL;
+	g_autoptr(GRWLockReaderLocker) locker = g_rw_lock_reader_locker_new (&priv->rw_lock);
+
+	result = as_cache_get_components_by_bundle_id (priv->cache, kind, bundle_id, &tmp_error);
+	if (result == NULL) {
+		g_warning ("Unable find components offering the bundle in session cache: %s", tmp_error->message);
+		return g_ptr_array_new_with_free_func (g_object_unref);
+	}
+
+	return result;
+}
+
```

Comment:
this needs a GIR wrapper, like the one below too... But I could add that as a follow-up commit, if you don't want to dive into writing binding code.

Reply:
Please, I'd appreciate it if you could look into this later.

Reply:
It's quite trivial, so, no problem :-)